internal/compose/compose.go

```diff
@@ -340,7 +343,38 @@ func (p *Project) runDockerComposeCmd(opts dockerComposeOptions) error {
 	return cmd.Run()
 }
 
+func (p *Project) dockerComposeVersion() (*semver.Version, error) {
+
+	var b bytes.Buffer
+
+	args := []string{
+		"version",
+		"--short",
+	}
+	if err := p.runDockerComposeCmd(dockerComposeOptions{args: args, stdout: &b}); err != nil {
+		return nil, errors.Wrap(err, "running Docker Compose version command failed")
+	}
+	dcVersion := b.String()
+	ver, err := semver.NewVersion(dcVersion)
+	if err != nil {
+		return nil, errors.Wrapf(err, "docker compose version not a valid semver %s", dcVersion)
+	}
+	return ver, nil
+}
+
 // ContainerName method the container name for the service.
 func (p *Project) ContainerName(serviceName string) string {
-	return fmt.Sprintf("%s_%s_1", p.name, serviceName)
+	if p.dcMajorVersion == "" {
```

Comment:
It might be cleaner to keep the `ContainerName` function as simple as possible:

```
if p.dockerComposeV1 {
   return fmt.Sprintf("%s_%s_1", p.name, serviceName)
}
return fmt.Sprintf("%s-%s-1", p.name, serviceName)
```

Then, the logic determining `p.dockerComposeV1` can be moved to the `NewProject`.

`dockerComposeV1` can be bool, as the [Compose V2](https://docs.docker.com/compose/cli-command/) seems to deprecate V1 in the future.